File: order-service/src/main/java/com/factoryx/order/outbox/OutboxEvent.java
package com.factoryx.order.outbox;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.AccessLevel;

import java.time.Instant;

@Entity
@Table(name = "outbox")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class OutboxEvent {

    @EmbeddedId
    @AttributeOverride(name = "value", column = @Column(name = "id"))
    private OutboxEventId id;

    @Embedded
    private AggregateType aggregateType;

    @Column(nullable = false)
    private String aggregateId;

    @Embedded
    @AttributeOverride(name = "value", column = @Column(name = "type"))
    private EventType type;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String payload;

    @Column(nullable = false)
    private Instant createdAt;

    public OutboxEvent(AggregateType aggregateType, String aggregateId, EventType type, String payload) {
        this.id = OutboxEventId.generate();
        this.aggregateType = aggregateType;
        this.aggregateId = aggregateId;
        this.type = type;
        this.payload = payload;
        this.createdAt = Instant.now();
    }
}

File: inventory-service/src/main/kotlin/com/factoryx/inventory/outbox/OutboxEvent.kt
package com.factoryx.inventory.outbox

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant
import java.util.*

@Entity
@Table(name = "outbox")
class OutboxEvent(
    @Id
    val id: UUID = UUID.randomUUID(),

    @Column(nullable = false)
    val aggregateType: String,

    @Column(nullable = false)
    val aggregateId: String,

    @Column(name = "type", nullable = false)
    val type: String,

    @Column(columnDefinition = "TEXT", nullable = false)
    val payload: String,

    @Column(nullable = false)
    val createdAt: Instant = Instant.now()
)
